model/textparse/openmetricsparse.go

```diff
@@ -212,9 +212,26 @@ func (p *OpenMetricsParser) Exemplar(e *exemplar.Exemplar) bool {
 	return true
 }
 
-// CreatedTimestamp returns nil as it's not implemented yet.
-// TODO(bwplotka): https://github.com/prometheus/prometheus/issues/12980
+// For OpenMetrics text, the created timestamp is exposed as it was an extra metric.
+// The difference between an usual metric and the created timestamp is that for
+// the created timestamp, the metric name ends with "_created".
+//
+// Created timestamps are always exposed right after the metric they are
```

Comment:
> right after

Is that part of the OpenMetrics spec? I think in practice I have only seen `_created` as the last metric in a MetricFamily, but I haven't read anything in the spec that requires that.

Reply:
You're correct, I meant MetricPoint, sorry!

I agree that all of the samples show `_created` as the last line, but that doesn't seem enforced so I think it is possible and valid OpenMetrics for `_created` to come first.

Reply:
Yeah, it might be worth getting an opinion from someone more involved in OpenMetrics than myself as I agree, it would be nice not to have to worry about the order!

Reply:
I'm not more involved in OpenMetrics, but I think it's totally fine that an experimental feature relies on this accidental property.

Reply:
Yea, the order of _created is not guaranteed to be the last. But also I don't understand you worries @ArthurSens -- if it's last we already have the problem of having to "buffer" previous samples related to the _created  metric which is potentially at the end. So it does not matter if it's end, middle or first. We still have to buffer samples (and corresponding series, which should be the same) no matter what, so we can add potential 0 sample OR in future add it to metadata.

Right now CreatedTimestamp() returns number ONLY when parser hits _created metric, which is clearly not enough for OM as we already written sample for related series no? We have to change bit this logic to e.g. buffer series in (OpenMetrics parsing only).

To me that means e.g. having different `append` function all together for OM, and that's fine. It **should** motivate, heavy changes to newer OM protocol to fix this.

Reply:
I'm not 100% sure yet(doing research at the moment), but I believe all official client libraries that offer OpenMetrics support today expose `_created` lines after the related metric. If I can confirm this, and seeing that this feature is experimental, could we rely on this order for this PR?

Expecting different order for the _created lines will make the code a lot more complex, we could make reviews easier if we split this up in different work streams :)